File: go/exercises/basics/reverse.go
package basics

// Reverse will return the provided word in reverse
// order. Eg:
//
//   Reverse("cat") => "tac"
//   Reverse("alphabet") => "tebahpla"
//
func Reverse(word string) string {
	res := make([]byte, len(word))
	lw := len(word)
	l := len(word) - 1
	for i := 0; i < lw; i++ {
		res[i] = word[l-i]
	}
	return string(res)
}

File: javascript/exercises/basics/reverse.js
// Reverse will return the provided word in reverse
// order. Eg:
//
//   Reverse("cat") => "tac"
//   Reverse("alphabet") => "tebahpla"
//
// Refrain using the built in the revese() method

function reverse(word) {
  let letters = word.split("");
  let wordLength = letters.length;
  let result = [];
  console.log(word + "\n");
  for (let i = 0; i < letters.length; i++) {
    result += letters[wordLength - i - 1];
  }

  return result;
}

export default reverse;
